ui/src/main/java/edu/wpi/grip/ui/codeGeneration/java/TPipeline.java

```diff
@@ -0,0 +1,149 @@
+package edu.wpi.grip.ui.codegeneration.java;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.wpi.grip.core.Connection;
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.ui.codegeneration.TemplateMethods;
+
+
+public class TPipeline {
```

Comment:
I see a lot of logic in this file and in TemplateMethods that are dealing with types, sometimes in a language-specific way (e.g. TemplateMethods::parseSocketType). This is fragile and will complicate efforts to generate code for multiple languages.

Different languages will necessarily need different language-specific declarations in order to represent sockethint values in a pipeline. It would greatly ease the burden of supporting different languages if the type information provided to templates were the same for every language (e.g. the sockethint types themselves), and the template for each language defined a macro that converts the common abstract type into the appropriate language-specific declaration.